pkg/controller/backup_sync_controller.go

```diff
@@ -107,4 +125,38 @@ func (c *backupSyncController) run() {
 			}
 		}
 	}
+
+	err = deleteUnused(c, cloudBackups)
+	if err != nil {
+		c.logger.WithError(errors.WithStack(err)).Error("Error deleting unused backup from Kubernetes")
+	}
+}
+
+// deleteUnused deletes backups from Kubernetes if they do not exist in object storage
+func deleteUnused(c *backupSyncController, cloudBackups map[string]string) error {
+	// Backups from Kubernetes
+	backups, err := c.backupLister.Backups(c.namespace).List(labels.Everything())
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	if len(backups) == 0 {
+		return nil
+	}
+
+	// For each backup item in Kubernetes, verify if has a corresponding backup in the cloud object storage. If not, delete it.
+	var deleteErrors error
+	for _, backup := range backups {
+		if _, ok := cloudBackups[backup.Name]; !ok {
+			c.client.Backups(backup.Namespace).Delete(backup.Name, &metav1.DeleteOptions{})
+			if err != nil {
+				deleteErrors = multierror.Append(deleteErrors, errors.WithStack(err))
```

Comment:
This is just an FYI given my previous comment recommending we not return errors:

When we're dealing with multiple errors, we usually either return a slice of errors, or use `k8s.io/apimachinery/pkg/util/errors` and the `NewAggregate()` func, which will aggregate them into a single error.  We could discuss whether using the hashicorp library would be a better solution in general, but otherwise, I think we'd prefer not to pull in an additional library.